File: Lex+Yacc-b.c
/* LEX FILE */
%{
#include "y.tab.h"
%}

%%
[a-zA-Z][a-zA-Z0-9]*   { return 1; }
.                      { return 0; }
%%

/* YACC FILE */
%{
#include<stdio.h>
%}

%%

S : ID { printf("Valid Variable\n"); }
  ;

%%

int main() {
    yyparse();
    return 0;
}

int yyerror() {
    printf("Invalid Variable\n");
    return 0;
}

File: Lex+Yacc-b.java
import java.util.*;

public class VariableCheck {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter variable: ");
        String var = sc.next();

        if(var.matches("[a-zA-Z][a-zA-Z0-9]*"))
            System.out.println("Valid Variable");
        else
            System.out.println("Invalid Variable");
    }
}